rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -3609,5 +3615,74 @@ public Response listStudentSubjectChoices(@PathParam("STUDENTID") Long studentId
     return Response.ok(variableList).build();
   }
 
+  @Path("/students/{STUDENTID:[0-9]*}/studentCard")
+  @GET
+  @RESTPermit(handling = Handling.INLINE)
+  public Response findStudentCard(@PathParam("STUDENTID") Long studentId) {
+    Student student = studentController.findStudentById(studentId);
+    Status studentStatus = checkStudent(student);
+    
+    if (studentStatus != Status.OK)
+      return Response.status(studentStatus).build();
+    
+    User loggedUser = sessionController.getUser();
 
+      if (!loggedUser.getId().equals(student.getId()) && !loggedUser.hasRole(Role.ADMINISTRATOR)) {
```

Comment:
What about administrator access? Isn't this endpoint for students themselves only?